opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -1,60 +1,122 @@
-use opentelemetry::logs::{LogRecord, Logger, LoggerProvider, Severity};
+use opentelemetry::{
+    logs::{AnyValue, LogRecord, Logger, LoggerProvider, Severity},
+    Key,
+};
 use std::borrow::Cow;
-use tracing_core::Level;
+use tracing_core::{Level, Metadata};
+use tracing_log::NormalizeEvent;
 use tracing_subscriber::Layer;
 
 const INSTRUMENTATION_LIBRARY_NAME: &str = "opentelemetry-appender-tracing";
+// Start by "log."
+const LOG_METADATA_ATTRIBUTES: &[&str] = &[
+    "name",
+    "target",
+    "module.path",
+    "file.path",
+    "file.name",
+    "file.line",
+    "module_path",
+    "file",
+    "line",
+];
 
 /// Visitor to record the fields from the event record.
-struct EventVisitor<'a> {
-    log_record: &'a mut LogRecord,
+#[derive(Default)]
+struct EventVisitor {
+    log_record_attributes: Vec<(Key, AnyValue)>,
+    log_record_body: Option<AnyValue>,
 }
 
-impl<'a> tracing::field::Visit for EventVisitor<'a> {
+fn is_metadata(field: &str) -> bool {
+    if let Some(log_field) = field.strip_prefix("log.") {
+        for log_metadata_field in LOG_METADATA_ATTRIBUTES {
+            if *log_metadata_field == log_field {
+                return true;
+            }
+        }
+    }
+    false
+}
+
+fn get_filename(filepath: &str) -> &str {
+    if let Some((_, filename)) = filepath.rsplit_once('/') {
+        return filename;
+    }
+    if let Some((_, filename)) = filepath.rsplit_once('\\') {
+        return filename;
+    }
+    filepath
+}
```

Comment:
Can we use `std::path::Path` ?

Reply:
I though about it but I was not sure because std::path::Path behaviour is dependant on the host OS.

So two cases where I think it can cause problem:

1. Cross-compilation : If we compile on Linux the Windows binary, do std::path::Path will try splitting the path with '\' instead of '/'? Because logically the original path of source files will be with '/'.
2. Bare Metal (No OS) : Do std::path will be available?

But I'm not against using std::path::Path if people judge it is not necessary to handle these cases. :)